Pilha.java

```diff
@@ -0,0 +1,75 @@
+public class Pilha {
+
+    public Object[] pilha; 
+    public int posicaoPilha;
+    public int tamMax;
+
+    public Pilha() {
+        this.posicaoPilha = -1;
+        this.pilha = new Object[10];
+        tamMax = 10;
+    }
+
+    public boolean pilhaVazia() {
+        //isEmpty
+        if (this.posicaoPilha == -1) {
+            return true;
+        }
+        return false;
+    }
+    
+    public boolean pilhaCheia() {
+    	//isFull
+    	if (this.posicaoPilha == tamMax){ 
+    		return true;
+    	} else {
+    		return false;
+    	}
+    }
+    
+
+    public int tamanho() {
+        //is
+        if (this.pilhaVazia()) {
+            return 0;
+        }
+        return this.posicaoPilha; 
```

Comment:
Essa linha apresenta um problemas. O valor de posicaoPilha está errado, como essa variável inicia com o valor -1, logo quanto tiver apenas um elemento na pilha o retorno será 0 e não 1. Se tiver 2 elementos o retorno será 1 e não 2. Para corrigir isso adicione +1 no final do retorno: return this.posicaoPilha+1;